ext/node/ops/ipc.rs

```diff
@@ -0,0 +1,179 @@
+// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
+
+use std::cell::RefCell;
+use std::io;
+use std::os::fd::FromRawFd;
+use std::os::fd::RawFd;
+use std::rc::Rc;
+
+use deno_core::error::bad_resource_id;
+use deno_core::error::AnyError;
+use deno_core::op2;
+use deno_core::serde_json;
+use deno_core::CancelFuture;
+use deno_core::CancelHandle;
+use deno_core::OpState;
+use deno_core::RcRef;
+use deno_core::ResourceId;
+use tokio::net::UnixStream;
+
+struct IpcPipe {
+  // Better name?
+  inner: UnixStream,
+}
+
+impl IpcPipe {
+  fn new(fd: RawFd) -> Result<Self, std::io::Error> {
+    Ok(Self {
+      inner: UnixStream::from_std(unsafe {
+        std::os::unix::net::UnixStream::from_raw_fd(fd)
+      })?,
+    })
+  }
+
+  async fn write(self: Rc<Self>, data: &[u8]) -> Result<usize, std::io::Error> {
+    let mut offset = 0;
+    loop {
+      self.inner.writable().await?;
+      match self.inner.try_write(&data[offset..]) {
+        Ok(n) => {
+          offset += n;
+          if offset >= data.len() {
+            return Ok(offset);
+          }
+        }
+        Err(ref e) if e.kind() == io::ErrorKind::WouldBlock => {
+          continue;
+        }
+        Err(e) => return Err(e),
+      }
+    }
+  }
+
+  async fn read(&self, data: &mut [u8]) -> Result<usize, std::io::Error> {
+    loop {
+      self.inner.readable().await?;
+      match self.inner.try_read(&mut data[..]) {
+        Ok(n) => {
+          return Ok(n);
+        }
+        Err(ref e) if e.kind() == io::ErrorKind::WouldBlock => {
+          continue;
+        }
+        Err(e) => return Err(e),
+      }
+    }
+  }
+}
+
+struct IpcJsonStream {
+  pipe: Rc<IpcPipe>,
+  buffer: RefCell<Vec<u8>>,
+  cancel: CancelHandle,
+}
+
+impl IpcJsonStream {
+  fn new(fd: RawFd) -> Result<Self, std::io::Error> {
+    Ok(Self {
+      pipe: Rc::new(IpcPipe::new(fd)?),
+      buffer: RefCell::new(Vec::new()),
+      cancel: CancelHandle::default(),
+    })
+  }
+
+  async fn read(self: Rc<Self>) -> Result<Vec<serde_json::Value>, AnyError> {
+    let mut buf = [0u8; 1024]; // TODO: Use a single growable buffer.
+    let mut msgs = Vec::new();
+    loop {
+      let n = self.pipe.read(&mut buf).await?;
+
+      let read = &buf[..n];
+      let mut chunk_boundary = 0;
+
+      for byte in read {
+        if *byte == b'\n' {
+          let chunk = &read[..chunk_boundary];
+          self.buffer.borrow_mut().extend_from_slice(chunk);
+
+          chunk_boundary = 0;
+          if chunk.is_empty() {
+            // Last chunk.
+            break;
+          }
+          msgs.push(serde_json::from_slice(&self.buffer.borrow())?);
+          self.buffer.borrow_mut().clear();
+        } else {
+          chunk_boundary += 1;
+        }
+      }
```

Comment:
Definitely needs a test.

Method should be named `read_msgs`. 

For reviewers: This implements the JSON serialization format in Node IPC in Rust to avoid entering JS for each read.